x/supplier/types/message_submit_proof_test.go

```diff
@@ -7,34 +7,93 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"github.com/pokt-network/poktroll/testutil/sample"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
 )
 
-// TODO(@bryanchriswhite): Add unit tests for message validation when adding the business logic.
-
 func TestMsgSubmitProof_ValidateBasic(t *testing.T) {
+	testService := &sharedtypes.Service{Id: "svc01"}
+	testClosestMerkleProof := []byte{1, 2, 3, 4}
+
 	tests := []struct {
-		name string
-		msg  MsgSubmitProof
-		err  error
+		desc        string
+		msg         MsgSubmitProof
+		expectedErr error
 	}{
 		{
-			name: "invalid address",
+			desc: "application bech32 address is invalid",
+			msg: MsgSubmitProof{
+				SupplierAddress: sample.AccAddress(),
+				SessionHeader: &sessiontypes.SessionHeader{
+					ApplicationAddress:      "not_a_bech32_address",
+					Service:                 testService,
+					SessionStartBlockHeight: 0,
+					SessionId:               "mock_session_id",
+					SessionEndBlockHeight:   5,
+				},
+				Proof: testClosestMerkleProof,
+			},
+			expectedErr: sdkerrors.ErrInvalidAddress.Wrapf(
```

Comment:
See the messages above about defining a type for this specific to the supplier module

Reply:
This must match whatever error variable we decide to go with in the above. Let's continue the conversation there (https://github.com/pokt-network/poktroll/pull/328#discussion_r1466082964).